File: C/exercise/convertDateToBinary.c

#include "convertDateToBinary.h"
#include <stdio.h>

static void intToBinary(int num, char *binary) {
	char temp[32];
	int i = 0;
	while (num > 0) {
		temp[i] = (num % 2) + '0';
		num /= 2;
		i++;
	}
	for (int j = 0; j < i; j++) {
		binary[j] = temp[i - 1 - j];
	}
	binary[i] = '\0';
}

char *convertDateToBinary(char *date)
{
	static char binaryDate[200];
	int year, month, day;

	(void)sscanf(date, "%d-%d-%d", &year, &month, &day);

	char yearBinary[32], monthBinary[32], dayBinary[32];
	intToBinary(year, yearBinary);
	intToBinary(month, monthBinary);
	intToBinary(day, dayBinary);

	(void)sprintf(binaryDate, "%s-%s-%s", yearBinary, monthBinary, dayBinary);

	return binaryDate;
}

File: Dart/lib/convertDateToBinary.dart
class Solution {
  String convertDateToBinary(String date) {
    List<String> parts = date.split('-');
    String yearBinary = int.parse(parts[0]).toRadixString(2);
    String monthBinary = int.parse(parts[1]).toRadixString(2);
    String dayBinary = int.parse(parts[2]).toRadixString(2);
    return '$yearBinary-$monthBinary-$dayBinary';
  }
}
